### PR TITLE
feat: Integrate pydantic validation into supervisor

### PR DESCRIPTION
**Pull request by robinholzi**
# Motivation

The supervisor logic can be streamlined by moving the validation logic into the pydantic models.

# Changes

- Add additional validation to pydantic models
- Adjust tests to work on actual "minimal configs"; the dicts previously wouldn't have passed both the old jsonschemas as well as the new pydantic schemas. Adding global fixtures for there models.

### CODE REVIEW

File: modyn/tests/supervisor/internal/test_supervisor.py [hunk 1]
```diff
@@ -172,163 +114,84 @@ def get(self, some_type, pipeline_id):  # pylint: disable=unused-argument
...
 
     return supervisor
 
 
-def test_initialization() -> None:
-    get_non_connecting_supervisor()  # pylint: disable=no-value-for-parameter
-
-
-@patch.object(Supervisor, "__init__", noop_constructor_mock)
-def test__validate_training_options():
```
Comment (robinholzi) [inline modyn/tests/supervisor/internal/test_supervisor.py:192]: There's no need for those trivial tests, pydantic models cannot even be constructed with violating those constraints.

File: modyn/config/schema/pipeline.py [hunk 2]
```diff
@@ -75,20 +77,22 @@ class PipelineModelStorageConfig(BaseModel):
 class PresamplingConfig(BaseModel):
     """Config for the presampling strategy of CoresetStrategy. If missing, no presampling is applied."""
 
-    strategy: Literal["Random", "RandomNoReplacement", "LabelBalanced", "TriggerBalanced", "No"] = Field(
-        description="Strategy used to presample the data."
-    )
-    ratio: float = Field(
+    strategy: Literal[
```
Comment (robinholzi) [inline modyn/config/schema/pipeline.py:80]: This list wasn't complete in the JsonSchema

File: modyn/supervisor/internal/supervisor.py [hunk 3]
```diff
@@ -78,116 +79,63 @@ def init_cluster_connection(self) -> None:
...
-
-    # pylint: disable=too-many-branches
-    def _validate_training_options(self, training_config: dict) -> bool:
-        is_valid = True
-        batch_size = training_config["batch_size"]
-        dataloader_workers = training_config["dataloader_workers"]
-        strategy = training_config["selection_strategy"]["name"]
-        initial_model = training_config["initial_model"]
-
-        if training_config["gpus"] != 1:
```
Comment (robinholzi) [inline modyn/supervisor/internal/supervisor.py:116]: In pydantic models

**Comment by MaxiBoether** [comment]
I will wait for #409 to be merged (hopefully soon) before merging this to avoid conflicts in #409 :)

**Comment by robinholzi** [comment]
@MaxiBoether Can you reopen the PR, I think collaborators cannot do that. Deleting a merged base branch apparently closes all related PRs.
Could you maybe also adjust the repo settings so that every (squash) merge automatically deletes the branch on GitHub? Then we won't run into those issues again :)

**Comment by MaxiBoether** [comment]
I cannot reopen the PR because the target branch of this PR `robinholzi/fix/local-grpc-build-and-integration-tests` has been deleted. I cannot change the target branch. As the author of the PR, you should be able to change the target branch, though? I did this before with one of my PRs but for this PR it does not offer the option. Otherwise, can you open a new PR into main maybe?

**Pull request closed without merging**